pkg/proxy/proxy_test.go

```diff
@@ -599,12 +600,69 @@ func (s *TestProxySuite) TestGetWorkspaceContext() {
 			expectedPath:      "/api/pods",
 			expectedErr:       "",
 		},
+		"no workspace context but plugins in kube api portion": {
+			path:              "/api/plugins/something",
+			expectedWorkspace: "",
+			expectedPath:      "/api/plugins/something",
+			expectedErr:       "",
+		},
 		"workspace instead of workspaces": {
 			path:              "/workspace/myworkspace/api",
 			expectedWorkspace: "",
 			expectedPath:      "/workspace/myworkspace/api",
 			expectedErr:       "",
 		},
+		"valid workspace context with route": {
+			path:              "/plugins/tekton-results/workspaces/myworkspace/api",
+			expectedWorkspace: "myworkspace",
+			expectedPath:      "/api",
+			expectedErr:       "",
+			expectedProxy:     "tekton-results",
+		},
+		"valid workspace context with route plus plugin in kube api portion": {
+			path:              "/plugins/tekton-results/workspaces/myworkspace/api/plugins/something",
+			expectedWorkspace: "myworkspace",
+			expectedPath:      "/api/plugins/something",
+			expectedErr:       "",
+			expectedProxy:     "tekton-results",
+		},
+		// the pre-plugin testcases all assumed /workspace would exists or /api/... exists; should we assume /plugins/<plugin name>/workspace or /plugins/<plugin name/api/... exists?
```

Comment:
Not sure I understand what you mean here. If you mean that it doesn't check anything before continuing with the request then yes, it just processes the request according to the context. The request would fail later in the flow if there's something wrong. Is that what you meant?

Reply:
yes that is what I was getting at .... thanks for confirming

I can remove the comments with my next update to the PR